examples/cozmo_face_mirror.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2016 Anki, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License in the file LICENSE.txt or at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+'''Display Cozmo's camera feed back on his face (like a mirror)
+'''
+
+import sys
+import time
+
+try:
+    from PIL import Image
+except ImportError:
+    sys.exit("Cannot import from PIL: Do `pip3 install --user Pillow` to install")
+
+try:
+    import numpy as np
+except ImportError:
+    sys.exit("Cannot import numpy: Do `pip3 install --user numpy` to install")
+
+
+import cozmo
+
+
+def get_in_position(robot):
+    '''If necessary, Move Cozmo's Head and Lift to make it easy to see Cozmo's face'''
+    if (robot.lift_height.distance_mm > 45) or (robot.head_angle.degrees < 40):
+        with robot.perform_off_charger():
+            robot.set_lift_height(0.0).wait_for_completed()
+            robot.set_head_angle(cozmo.robot.MAX_HEAD_ANGLE).wait_for_completed()
+
+
+def calc_pixel_threshold(image):
+    '''Calculate a pixel threshold based on the image
+
+    Anything brighter than this will be shown on (light blue)
+    Anything darker will be shown off (black)
+    '''
+
+    # Convert image to gray scale
+    grayscale_image = image.convert('L')
+
+    # Calculate the mean (average) value
+    mean_value = np.mean(grayscale_image.getdata())
+    return mean_value
+
+
+def cozmo_face_mirror(robot):
+
+    robot.camera.image_stream_enabled = True
+
+    face_dimensions = cozmo.oled_face.SCREEN_WIDTH, cozmo.oled_face.SCREEN_HALF_HEIGHT
+
+    while True:
+        duration_s = 0.1  # time to display each camera frame on Cozmo's face
+
+        latest_image = robot.world.latest_image
+
+        if latest_image is not None:
+            # scale the camera image down to fit on Cozmo's face
+            resized_image = latest_image.raw_image.resize(face_dimensions,
+                                                          Image.BICUBIC)
+
+            # Flip the image left/right so it displays mirrored
+            resized_image = resized_image.transpose(Image.FLIP_LEFT_RIGHT)
+
+            # calculate the pixel threshold for this image
+            pixel_threshold = calc_pixel_threshold(resized_image)
+
+            # convert the image to the format to display on Cozmo's face
```

Comment:
To make this more readable, should start each of these 2 sentences with a capital and end with a period.